R/backport_linter.R

```diff
@@ -22,7 +46,10 @@ backport_linter <- function(r_version = getRversion()) {
     all_names <- xml2::xml_text(all_names_nodes)
 
     # guaranteed to include 1 by early return above; which.min fails if all TRUE (handled by nomatch)
-    needs_backport_names <- backports[1:(match(FALSE, r_version < names(backports), nomatch = length(backports)) - 1L)]
+    needs_backport_names <- head(
+      backports,
+      match(FALSE, r_version < R_system_version(names(backports)), nomatch = length(backports)) - 1L
```

Comment:
tricky code... extract as a helper with doc/comments explaining what it's doing / why not use something more obvious

Reply:
I've refactored this now to use logical indexing and changed the variable name. Does that make it more understandable?

Reply:
Didn't see the `match()` thing was just copied from earlier (in fact IIRC I am the author of it 😃), but was talking about that, not so much `head()`.

definitely much cleaner now... i wonder why we didn't do that before 🤔